nautilus_core/common/src/logging.rs

```diff
@@ -481,9 +585,9 @@ impl Logger {
 
             match event {
                 LogEvent::Flush => {
-                    Self::flush_stderr(&mut err_buf);
-                    Self::flush_stdout(&mut out_buf);
-                    file_buf.as_mut().map(Self::flush_file);
+                    stdout_writer.flush();
```

Comment:
This should not be needed anymore since the writers should flush when dropped. When the a Flush log even is received logic should simply break out of the loop.